### PR TITLE
Resizing, pre-processing and saving to tfrecord file

### PR DESCRIPTION
**Pull request by chamathabeysinghe**

### CODE REVIEW

File: dataset/resize.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        fps = video.get(cv2.CAP_PROP_FPS)
+        fourcc = cv2.VideoWriter_fourcc(*'XVID')
+        out = cv2.VideoWriter(output_file, fourcc, fps, size)
+
+        # iterate through each frame
+        while video.isOpened():
+            ret, frame = video.read()
+            if not ret:
+                break
+            # use built in resize function - need to be changed
```
Comment (ThejanW) [inline dataset/resize.py]: we should first select what is the greater value out of height and width, then accordingly resize the frame including padding, where it's needed.

File: dataset/resize.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                        default=244,
+                        type=int,
+                        metavar='WIDTH',
+                        dest='width',
+                        help='use WIDTH as width of a frame')
+
+    args = parser.parse_args()
+
+    video_resizer = VideoResizer(args.source, args.output)
+    video_resizer.resize_all((args.width, args.height))
```
Comment (ThejanW) [inline dataset/resize.py]: nice work :+1:

File: dataset/sampling.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            writer.write(serialized)
+
+        if verbose:
+            print "Writing {} done!".format(result_tf_file)
+
+    def process(self, length=32, skip=1):
+        X, Y = self.slice_all()
+        # problem: image type matters if bit 8 or bit 32 image
+        # https://docs.opencv.org/2.4/modules/imgproc/doc/miscellaneous_transformations.html
+        x_reshaped = np.divide(X.reshape(X.shape[0], X.shape[1] * X.shape[2] * X.shape[3]), 100.0)
```
Comment (ThejanW) [inline dataset/sampling.py]: here, original shape of X is (num_vids, num_frames_in_vid, frame_h, frame_w) & shape of Y is (num_vids, num_frames_in_vid, 2, frame_h, frame_w) right?

Comment (ThejanW) [reply dataset/sampling.py]: This is wrong. We have to divide it by 50 and subtract 1(to bring it to -1, +1 interval from 0, 100 interval), not the entire thing by 100.

File: dataset/sampling.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    parser.add_argument('-k', '--skip',
+                        default=3,
+                        type=int,
+                        metavar='SKIP',
+                        dest='skip',
+                        help='use SKIP as number of frames between two selected frames')
+
+    args = parser.parse_args()
+
+    sample = Sample(args.source, args.output, args.length, args.skip)
```
Comment (ThejanW) [inline dataset/sampling.py]: nicely done :100:

File: dataset/resize.py (hunk 1 above)
Comment (chamathabeysinghe) [reply dataset/resize.py]: Yeah.. That's why I mentioned need to be changed. Will change this soon.

File: dataset/sampling.py (hunk 3 above)
Comment (chamathabeysinghe) [reply dataset/sampling.py]: Yes shape is correct. 
Will change the division forgot L doesn't have negative range

**Comment by chamathabeysinghe** [comment]
@ThejanW @Lahiru-J @merovingienne changes made. review

**Pull request merged**